res/controllers/Pioneer-DDJ-200-scripts.js

```diff
@@ -36,6 +48,15 @@ DDJ200.init = function() {
     }, true);
 };
 
+DDJ200.listemHotcues = function(vgroup) {
+    for (var i = 1; i <= 8; i++) {
+        // run updateDeckLeds after every hotcue update
+        engine.makeConnection(vgroup, "hotcue_" + i + "_enabled", function(ch, vgroup) {
+            DDJ200.updateDeckLeds(vgroup);
```

Comment:
This looks like it sends a MIDI message for every hotcue when only one of them updates? That's inefficient.

Reply:
I refactored this whole thing to switch only leds that actually changed